R/genius_url.R

```diff
@@ -18,67 +18,92 @@
 #' @importFrom stringr str_detect str_extract
 #' @importFrom readr read_lines
 
-genius_url <- function(url, info = "title") {
- session <- html_session(url)
- # Clean the song lyrics
- lyrics <- gsub(pattern = "<.*?>",
-                replacement = "\n",
-                html_node(session, ".lyrics")) %>%
-    read_lines() %>%
-    na.omit() %>%
-    str_replace_all("’", "'")
+genius_url <- function(url, info = "title")  {
+  # create a new session for scraping lyrics
+  session <- html_session(url)
 
-# Artist
-artist <- html_nodes(session, ".header_with_cover_art-primary_info-primary_artist") %>%
+  # Get Artist name
+  artist <- html_nodes(session, ".header_with_cover_art-primary_info-primary_artist") %>%
     html_text() %>%
     str_replace_all("\n", "") %>%
     str_trim()
 
-# Song title
-song_title <- html_nodes(session, ".header_with_cover_art-primary_info-title") %>%
+  # Get Song title
+  song_title <- html_nodes(session, ".header_with_cover_art-primary_info-title") %>%
     html_text() %>%
     str_replace_all("\n", "") %>%
     str_trim()
 
-# Convert to tibble
-lyrics <- tibble(artist = artist,
-            track_title = song_title,
-            lyric = lyrics)
+  # scrape the lyrics
+  lyrics <- # read the text from the lyrics class
+    html_node(session, ".lyrics") %>%
+    # trim white space
+    html_text(trim = TRUE) %>%
```

Comment:
removed the gsub call. This is slightly more robust, but requires more string cleaning. `rvest::html_text()` has some trouble with line endings